specs/PermissionManagement.md

```diff
@@ -0,0 +1,554 @@
+    CHECK_FAILURE(m_webView->add_WebMessageReceived(
+        Microsoft::WRL::Callback<ICoreWebView2WebMessageReceivedEventHandler>(
+            [this](ICoreWebView2* sender, ICoreWebView2WebMessageReceivedEventArgs* args)
+                -> HRESULT
+            {
+                wil::unique_cotaskmem_string source;
+                CHECK_FAILURE(args->get_Source(&source));
+                // Permission management APIs are only used on the app's
+                // permission management page.
+                if (source.get() != m_sampleUri)
+                {
+                    return S_OK;
+                }
+                wil::unique_cotaskmem_string message;
+                CHECK_FAILURE(args->TryGetWebMessageAsString(&message));
+                // The app's permission management page can provide a way for the
+                // end user to change permissions. For example, the page can have a
+                // `Set Permission` button that triggers a dialog, where the user
+                // specifies the desired origin, permission kind, and state.
+                if (wcscmp(message.get(), L"SetPermission") == 0)
+                {
+                    m_appWindow->RunAsync([this] { ShowSetPermissionDialog(); });
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_webMessageReceivedToken));
+}
+
+void ScenarioPermissionManagement::ShowSetPermissionDialog()
+{
+    PermissionDialog dialog(m_appWindow->GetMainWindow());
+    if (dialog.confirmed && m_webViewProfile6)
+    {
+        // Example: m_webViewProfile6->SetPermissionState(
+        //    COREWEBVIEW2_PERMISSION_KIND_GEOLOCATION,
+        //    L"https://example.com",
+        //    COREWEBVIEW2_PERMISSION_STATE_DENY,
+        //    SetPermissionStateCallback);
+        CHECK_FAILURE(m_webViewProfile6->SetPermissionState(
+            dialog.kind, dialog.origin.c_str(), dialog.state,
+            Callback<ICoreWebView2StagingSetPermissionStateCompletedHandler>(
+            [this](HRESULT error) -> HRESULT
+            {
+                // Reload the permission management page.
+                m_webView->Reload();
+                return S_OK;
+            })
+            .Get()));
+    }
+}
+```
+# API Details
+
+```
+[v1_enum]
+typedef enum COREWEBVIEW2_PERMISSION_KIND {
+
+  // Other permission kinds not shown.
+
+  /// Indicates permission to automatically download multiple files. Permission
+  /// is requested when multiple downloads are triggered at the same time.
+  COREWEBVIEW2_PERMISSION_KIND_AUTOMATIC_DOWNLOADS,
+
+  /// Indicates permission to edit files or folders on the device. Permission
+  /// is requested when developers use the [File System Access API](https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API)
+  /// to show the file or folder picker to the end user, and then request "readwrite"
+  /// permission for the user's selection.
+  COREWEBVIEW2_PERMISSION_KIND_FILE_EDITING,
```

Comment:
Name suggestion: sounds more like `MODIFY_FILE` access

Reply:
Please change to FILE_READ_WRITE to match DOM API https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle/requestPermission#:~:text=The%20requestPermission%20%28%29%20method%20of%20the%20FileSystemHandle%20interface,mode%20to%20query%20for.%20Options%20are%20as%20follows%3A